overlord/snapstate/snapstate.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package snapstate implements the manager and state aspects responsible for the installation and removal of snaps.
+package snapstate
+
+import (
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/i18n"
+	"github.com/ubuntu-core/snappy/overlord/state"
+	"github.com/ubuntu-core/snappy/snappy"
+)
+
+// Install returns a set of tasks for installing snap.
+// Note that the state must be locked by the caller.
+func Install(s *state.State, snap, channel string, flags snappy.InstallFlags) (*state.TaskSet, error) {
+	t := s.NewTask("install-snap", fmt.Sprintf(i18n.G("Installing %q"), snap))
+	t.Set("state", installState{
```

Comment:
to be honest  I find calling these *State a bit strange, unless it is what we are going to actually store in the state more permanently,   they are more like installParams (also wonder how we will pass them around when we have more than one task if we have to mostly ignore the change itself, just duplicating some of them in tasks?)